src/resources/bucket.ts

```diff
@@ -105,19 +142,38 @@ export class BucketResource extends SecureResource<BucketPermission> {
     req.setResource(resource);
 
     return new Promise<Resource>((resolve, reject) => {
-      resourceClient.declare(
-        req,
-        (error, response: ResourceDeclareResponse) => {
-          if (error) {
-            reject(fromGrpcError(error));
-          } else {
-            resolve(resource);
-          }
+      resourceClient.declare(req, (error, _: ResourceDeclareResponse) => {
+        if (error) {
+          reject(fromGrpcError(error));
+        } else {
+          resolve(resource);
         }
-      );
+      });
     });
   }
 
+  /**
+   * Register and start a bucket notification handler that will be called for all matching notification events on this bucket
+   *
+   * @param notificationType the notification type that should trigger the middleware, either 'write' or 'delete'
+   * @param notificationPrefixFilter the file name prefix that files must match to trigger a notification
+   * @param middleware handler middleware which will be run for every incoming event
+   * @returns Promise which resolves when the handler server terminates
+   */
+  on(
+    notificationType: string,
```

Comment:
if it is either write or delete, should we make this the arg type instead of string for better DX typing?